nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java

```diff
@@ -151,8 +158,10 @@ private static NessieClientBuilder<?> createNessieClientBuilder(String customBui
   }
 
   @Override
-  public void close() {
-    client.close();
+  public void close() throws IOException {
+    if (null != closeableGroup) {
```

Comment:
I know others don't have it but I feel that we could potentially still run into a case where `closeableGroup` hasn't been initialized because some preivous code in `initialize(...)` failed and if `close()` is called at some place, then it would throw a NPE